Update ES version to 6.6.0

henrywallace opened the pull request:
This is a naive attempt at upgrading the plugin to ES version 6.6.0

The changes are minimal, but I'm curious what opinions are on the removal of an `@Override`, and the removal of settings in `super`. The latter seems especially suspicious of being wrong.

Is there anything obvious going wrong here?

It appears, too, that after running the tests I see:

```
> Task :compileTestJava
Note: /home/ubuntu/diffeo/images/elasticsearch/elasticsearch-learning-to-rank/src/test/java/com/o19s/es/explore/ExplorerQueryTests.java uses or overrides a deprecated API.
```

and occassionally

```
> Task :integTestRunner
==> Test Info: seed=3D974279FB1A303; jvm=1; suites=7

Exception thrown by subscriber method onHeartbeat(com.carrotsearch.ant.tasks.junit4.events.aggregated.HeartBeatEvent) on subscriber com.carrotsearch.gradle.junit4.TestProgressLogger@5d88a632 when dispatching event: com.carrots
earch.ant.tasks.junit4.events.aggregated.HeartBeatEvent@30bc1695
java.lang.NullPointerException: Cannot get property 'className' on null object
        at org.codehaus.groovy.runtime.NullObject.getProperty(NullObject.java:60)
        at org.codehaus.groovy.runtime.InvokerHelper.getProperty(InvokerHelper.java:190)
        at org.codehaus.groovy.runtime.callsite.NullCallSite.getProperty(NullCallSite.java:46)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callGetProperty(AbstractCallSite.java:299)
        at com.carrotsearch.gradle.junit4.TestProgressLogger.onHeartbeat(TestProgressLogger.groovy:161)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at com.carrotsearch.ant.tasks.junit4.dependencies.com.google.common.eventbus.Subscriber.invokeSubscriberMethod(Subscriber.java:87)
        at com.carrotsearch.ant.tasks.junit4.dependencies.com.google.common.eventbus.Subscriber$SynchronizedSubscriber.invokeSubscriberMethod(Subscriber.java:144)
        at com.carrotsearch.ant.tasks.junit4.dependencies.com.google.common.eventbus.Subscriber$1.run(Subscriber.java:72)
        at com.carrotsearch.ant.tasks.junit4.dependencies.com.google.common.util.concurrent.MoreExecutors$DirectExecutor.execute(MoreExecutors.java:398)
        at com.carrotsearch.ant.tasks.junit4.dependencies.com.google.common.eventbus.Subscriber.dispatchEvent(Subscriber.java:67)
        at com.carrotsearch.ant.tasks.junit4.dependencies.com.google.common.eventbus.Dispatcher$PerThreadQueuedDispatcher.dispatch(Dispatcher.java:108)
        at com.carrotsearch.ant.tasks.junit4.dependencies.com.google.common.eventbus.EventBus.post(EventBus.java:212)
        at com.carrotsearch.ant.tasks.junit4.events.aggregated.AggregatingListener.slowHeartBeat(AggregatingListener.java:76)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at com.carrotsearch.ant.tasks.junit4.dependencies.com.google.common.eventbus.Subscriber.invokeSubscriberMethod(Subscriber.java:87)
        at com.carrotsearch.ant.tasks.junit4.dependencies.com.google.common.eventbus.Subscriber$SynchronizedSubscriber.invokeSubscriberMethod(Subscriber.java:144)
        at com.carrotsearch.ant.tasks.junit4.dependencies.com.google.common.eventbus.Subscriber$1.run(Subscriber.java:72)
        at com.carrotsearch.ant.tasks.junit4.dependencies.com.google.common.util.concurrent.MoreExecutors$DirectExecutor.execute(MoreExecutors.java:398)
        at com.carrotsearch.ant.tasks.junit4.dependencies.com.google.common.eventbus.Subscriber.dispatchEvent(Subscriber.java:67)
        at com.carrotsearch.ant.tasks.junit4.dependencies.com.google.common.eventbus.Dispatcher$PerThreadQueuedDispatcher.dispatch(Dispatcher.java:108)
        at com.carrotsearch.ant.tasks.junit4.dependencies.com.google.common.eventbus.EventBus.post(EventBus.java:212)
        at com.carrotsearch.ant.tasks.junit4.LocalSlaveStreamHandler$3.run(LocalSlaveStreamHandler.java:149)
HEARTBEAT J0 PID((?)): 2019-02-13T14:24:11, stalled for 10.0s at: <unknown>
```

nomoa commented on build.gradle:
```diff
@@ -51,10 +51,10 @@ dependencies {
   compile 'org.ow2.asm:asm-commons:5.0.4'
   compile 'org.ow2.asm:asm-tree:5.0.4'
-  compile 'org.elasticsearch:elasticsearch:6.5.4'
+  compile 'org.elasticsearch:elasticsearch:6.6.0'
```
lucene-expressions should be bumped to 7.5.1 because es 6.6.0 is compiled against lucene 7.5.1

henrywallace replied:
It doesn't look like 7.5.1 exists? https://mvnrepository.com/artifact/org.apache.lucene/lucene-expressions. Referencing https://raw.githubusercontent.com/elastic/elasticsearch/v6.6.0/buildSrc/version.properties as found from https://github.com/o19s/elasticsearch-learning-to-rank/issues/211#issuecomment-463125129, however, may imply 7.6? I've bumped it to 7.6 for now.

After building I get the errors
```
FAILURE: Build failed with an exception.

* What went wrong:
Could not determine the dependencies of task ':thirdPartyAudit'.
> Could not resolve all files for configuration ':runtime'.
   > Could not find org.apache.lucene:lucene-expressions:7.5.1.
     Searched in the following locations:
       - https://artifacts.elastic.co/maven/org/apache/lucene/lucene-expressions/7.5.1/lucene-expressions-7.5.1.pom
       - https://artifacts.elastic.co/maven/org/apache/lucene/lucene-expressions/7.5.1/lucene-expressions-7.5.1.jar
       - https://jcenter.bintray.com/org/apache/lucene/lucene-expressions/7.5.1/lucene-expressions-7.5.1.pom
       - https://jcenter.bintray.com/org/apache/lucene/lucene-expressions/7.5.1/lucene-expressions-7.5.1.jar
       - https://repo.maven.apache.org/maven2/org/apache/lucene/lucene-expressions/7.5.1/lucene-expressions-7.5.1.pom
       - https://repo.maven.apache.org/maven2/org/apache/lucene/lucene-expressions/7.5.1/lucene-expressions-7.5.1.jar
       - file:/home/ubuntu/.m2/repository/org/apache/lucene/lucene-expressions/7.5.1/lucene-expressions-7.5.1.pom
       - file:/home/ubuntu/.m2/repository/org/apache/lucene/lucene-expressions/7.5.1/lucene-expressions-7.5.1.jar
     Required by:
         project :
```

nomoa replied:
indeed, sorry for the confusion!

nomoa commented on src/main/java/com/o19s/es/ltr/ranker/ranklib/RankLibScriptEngine.java:
```diff
@@ -46,7 +46,7 @@
     private final LtrRankerParserFactory factory;
 
     public RankLibScriptEngine(Settings settings, LtrRankerParserFactory factory) {
-        super(settings);
```
if settings is no longer accepted in the parent constructor we should remove it from RankLibScriptEngine constructor as well.

nomoa commented on src/test/java/com/o19s/es/ltr/query/ValidatingLtrQueryBuilderTests.java:
```diff
@@ -108,7 +108,6 @@ protected boolean builderGeneratesCacheableQueries() {
         return false;
     }
 
-    @Override
```
Please see https://github.com/elastic/elasticsearch/pull/35967/commits/e882e624ad89bbaf279035907fda9d91295748d7
supportsBoostAndQueryName has been removed in favor of two distinct methods supportsBoost and supportsQueryName

adrpino commented:
Is there any way I could help with this?

henrywallace commented:
Thanks for the ping, I wasn't able to immediately get back on this. I'll try to address these changes in the next couple of days. Any further review/advice is welcome.

henrywallace commented:
@nomoa I believe I've addressed all of your comments, in a fixup commit, which we can squash with an ok.

There's one point on the correct version of lucene-expressions that is still unclear, however. For now I've bumped it to 7.6.0: https://github.com/o19s/elasticsearch-learning-to-rank/pull/213#discussion_r261027702.

The errors I was seeing previous I no longer am seeing when running the tests. Though it should be noted that I have not been able to stress test this plugin live yet.

jettro commented:
You can find the appropriate Lucene version here:
https://github.com/elastic/elasticsearch/blob/v6.6.0/buildSrc/version.properties

It shows they use Lucene 7.6.0

nomoa commented:
Thanks!
